src/crypto/openssl/key_pair.cpp

```diff
@@ -44,10 +45,17 @@ namespace crypto
     std::vector<std::pair<std::string, std::string>> r;
```

Comment:
That looks almost entirely like http::parse_query() except for the separator and the raw pointers. Could we maybe parametrise the implementation of http::parse_query() on the separator and use it in both places?

Reply:
Can't find `parse_query`, do you mean `parse_delimited_string`?

Reply:
Yes, I think it was called that in an earlier PR, but I'm obviously behind, sorry.